tests/integration/disambiguation/test_disambiguation_api.py

```diff
@@ -82,6 +83,42 @@ def test_save_curated_signatures_and_input_clusters(isolated_app, tmpdir):
     } in curated_signatures
 
 
+def test_save_sampled_signature_pairs(isolated_app, tmpdir):
+    TestRecordMetadata.create_from_file(__name__, '765515.json')
+    TestRecordMetadata.create_from_file(__name__, '765975.json')
+
+    curated_signatures_fd = tmpdir.join('curated_signatures.jsonl')
+    input_clusters_fd = tmpdir.join('input_clusters.jsonl')
+    sampled_pairs_fd = tmpdir.join('sampled_pairs.jsonl')
+
+    config = {
+        'DISAMBIGUATION_CURATED_SIGNATURES_PATH': str(curated_signatures_fd),
+        'DISAMBIGUATION_INPUT_CLUSTERS_PATH': str(input_clusters_fd),
+        'DISAMBIGUATION_SAMPLED_PAIRS_PATH': str(sampled_pairs_fd),
+        'DISAMBIGUATION_SAMPLED_PAIRS_SIZE': 1000,
+    }
+
+    with patch.dict(current_app.config, config):
+        save_curated_signatures_and_input_clusters()
+        save_sampled_pairs()
+
+    sampled_pairs = [json.loads(line) for line in sampled_pairs_fd.readlines()]
+    normalized_sampled_pairs = [
+        {
+            'same_cluster': sampled_pair['same_cluster'],
+            'signature_uuids': sorted(sampled_pair['signature_uuids']),
+        } for sampled_pair in sampled_pairs
+    ]  # XXX: so that the assertion doesn't depend on signature order.
```

Comment:
Not sure if this was the source of the flakiness I've observed: https://travis-ci.org/inspirehep/inspire-next/builds/391258552, https://travis-ci.org/inspirehep/inspire-next/builds/391378801, https://travis-ci.org/inspirehep/inspire-next/builds/391402413, https://travis-ci.org/inspirehep/inspire-next/builds/391402828.

Since it empirically failed 50% of the times it seemed sensible to look at something that could theoretically fail 50% of the times...

Reply:
I haven't seen it fail since, so I think I got it...